src/runtime/actor/mod.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{error::Error, fmt::Debug};
+
+use async_trait::async_trait;
+use futures::StreamExt;
+
+use self::context::ActorContext;
+
+/// Module containing the actor context
+pub mod context;
+/// Module containing event types
+pub mod envelope;
+/// Module containing actor error types
+pub mod error;
+/// Module containing the actor address handle
+pub mod handle;
+/// Module containing actor exit report types
+pub mod report;
+
+/// The actor trait, which defines a task that is managed by the runtime
+#[async_trait]
+pub trait Actor: Send + Sync + Sized {
+    /// Custom data that is passed to all actor methods
+    type Data: Debug + Send + Sync;
+    /// Custom error type that is returned by all actor methods
+    type Error: Error + Send + Sync;
+
+    /// Set this actor's name, primarily for debugging purposes
+    fn name(&self) -> String {
+        std::any::type_name::<Self>().into()
```

Comment:
why the allocation?

Reply:
Look what you have done. Now I've gone and had a `Cow`.